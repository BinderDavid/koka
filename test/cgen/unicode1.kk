val s0 = "x\u2603x (snowman)"
val s1 = "x\U01F4AAx (astral )"
val s2 = "x\U01D400x (math A )"

fun test1(s : string,ofs = 0) {
  println( "len: " + s.length.show + ", sub3: " + s.substr(ofs,3) + ", s: " + s)
}

fun main() {
  s0.test1
  s1.test1
  s2.test1
}
