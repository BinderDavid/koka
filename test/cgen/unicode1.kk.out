len: 13, sub3: x☃x, rev3: x☃x,
 s: x☃x (snowman), [78,2603,78,20,28,73,6E,6F,77,6D,61,6E,29]
len: 14, sub3: x💪, rev3: 💪x,
 s: x💪x (astral ), [78,1F4AA,78,20,28,61,73,74,72,61,6C,20,29]
len: 14, sub3: x𝐀, rev3: 𝐀x,
 s: x𝐀x (math A ), [78,1D400,78,20,28,6D,61,74,68,20,41,20,29]